pkg/instancegroups/drain.go

```diff
@@ -0,0 +1,595 @@
+	cmdutil.AddDryRunFlag(cmd)
+	return cmd
+}
+
+
+// SetupDrain populates some fields from the factory, grabs command line
+// arguments and looks up the node using Builder
+func (o *DrainOptions) SetupDrain(cmd *cobra.Command, args []string) error {
+	var err error
+	o.Selector = cmdutil.GetFlagString(cmd, "selector")
+
+	if len(args) == 0 && !cmd.Flags().Changed("selector") {
+		return cmdutil.UsageErrorf(cmd, fmt.Sprintf("USAGE: %s [flags]", cmd.Use))
+	}
+	if len(args) > 0 && len(o.Selector) > 0 {
+		return cmdutil.UsageErrorf(cmd, "error: cannot specify both a node name and a --selector option")
+	}
+	if len(args) > 0 && len(args) != 1 {
+		return cmdutil.UsageErrorf(cmd, fmt.Sprintf("USAGE: %s [flags]", cmd.Use))
+	}
+
+	o.DryRun = cmdutil.GetFlagBool(cmd, "dry-run")
+
+	if o.client, err = o.Factory.KubernetesClientSet(); err != nil {
+		return err
+	}
+
+	o.restClient, err = o.Factory.RESTClient()
+	if err != nil {
+		return err
+	}
+
+	o.nodeInfos = []*resource.Info{}
+	o.mapper, o.typer = o.Factory.Object()
+
+	cmdNamespace, _, err := o.Factory.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	builder := o.Factory.NewBuilder().
+		Internal().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		ResourceNames("nodes", args...).
+		SingleResourceType().
+		Flatten()
+
+	if len(o.Selector) > 0 {
+		builder = builder.LabelSelectorParam(o.Selector).
+			ResourceTypes("nodes")
+	}
+
+	r := builder.Do()
+
+	if err = r.Err(); err != nil {
+		return err
+	}
+
+	return r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.Mapping.Resource != "nodes" {
+			return fmt.Errorf("error: expected resource of type node, got %q", info.Mapping.Resource)
+		}
+
+		o.nodeInfos = append(o.nodeInfos, info)
+		return nil
+	})
+}
+
+// RunDrain runs the 'drain' command
+func (o *DrainOptions) RunDrain() error {
+	if err := o.RunCordonOrUncordon(true); err != nil {
+		return err
+	}
+
+	drainedNodes := sets.NewString()
+	var fatal error
+
+	for _, info := range o.nodeInfos {
+		var err error
+		if !o.DryRun {
+			err = o.deleteOrEvictPodsSimple(info)
+		}
+		if err == nil || o.DryRun {
+			drainedNodes.Insert(info.Name)
+			o.Factory.PrintSuccess(o.mapper, false, o.Out, "node", info.Name, o.DryRun, "drained")
+		} else {
+			fmt.Fprintf(o.ErrOut, "error: unable to drain node %q, aborting command...\n\n", info.Name)
+			remainingNodes := []string{}
+			fatal = err
+			for _, remainingInfo := range o.nodeInfos {
+				if drainedNodes.Has(remainingInfo.Name) {
+					continue
+				}
+				remainingNodes = append(remainingNodes, remainingInfo.Name)
+			}
+
+			if len(remainingNodes) > 0 {
+				fmt.Fprintf(o.ErrOut, "There are pending nodes to be drained:\n")
+				for _, nodeName := range remainingNodes {
+					fmt.Fprintf(o.ErrOut, " %s\n", nodeName)
+				}
+			}
+			break
+		}
+	}
+
+	return fatal
+}
+
+func (o *DrainOptions) deleteOrEvictPodsSimple(nodeInfo *resource.Info) error {
```

Comment:
rename this to `deleteOrEvictPodsSimple` b/c there is no more eviction?

Reply:
if you go with a flag approach, NM this.

Reply:
I didn't want to rename things since I basically wholesale copied this from the kubernetes repo